handout/forms/form-builder/form-builder_custom_validation.md

```diff
@@ -0,0 +1,102 @@
+# FormBuilder Custom Validation
+
+As useful as the built-in validators are, it is very useful to be able to include your own.
+Angular 2 allows you to do just that, with minimal effort.
+
+Let's assume we are using the same Login Form, but now we also want to test that our password has an exclamation mark somewhere in it.
+
+_app/login-form.component.ts_
+```ts
+function hasExclamationMark (input: FormControl) {
+  const hasExclamation = input.value.indexOf('!') >= 0;
+
+  return hasExclamation ? null : { needsExclamation: true };
+}
+
+
+// ...
+this.password = new FormControl('', [
+  Validators.required,
+  hasExclamationMark
+]);
+```
+
+I've created a simple function which takes the `FormControl` instance and returns null if everything is fine.
+If the test fails, it returns an object with an arbitrarily named property.
+The property name is what will be used for the `.hasError()` test.
+
+_app/login-form.component.ts_
+```html
+<!-- ... -->
+<div [hidden]="!password.hasError('needsExclamation')">
+  Your password must have an exclamation mark!
+</div>
+<!-- ... -->
+```
+
+### Predefined Parameters
+Having a custom validator check for exclamation marks might be helpful, but what if you need to check for some other form of punctuation?
```

Comment:
this could also be called validator factories ? but that's also a highly technical term. 

the more important note I had here was that this chapter doesn't really talk about Angular, instead, it talks about using advanced javascript principles with Angular.  

I would anticipate that a user trying to grasp the concepts of Angular 2 forms may end up having to now spend energy and effort on trying to grasp these javascript features, if they are not familiar with them ( and angular 2 in a sense, is aimed at non js devs ). 

would it make sense to put this chapter somewhere further down ? 
would it also make sense to reconsider the name ? 

i'm not sure if these are "predefined parameters" to me these are javascript functions which can build new validators on the fly.
and since it's a more advanced topic, not really angular 2 specific, maybe the name can also reflect that ?